src/pollers/mod.rs

```diff
@@ -60,21 +64,84 @@ pub struct ServerGatewayOptions {
 
     /// Timeout for long polls (polling of task queues)
     pub long_poll_timeout: Duration,
+
+    /// If specified, use TLS as configured by the [TlsCfg] struct
+    pub tls_cfg: Option<TlsConfig>,
+}
+
+/// Configuration options for TLS
+#[derive(Clone, Debug)]
+pub struct TlsConfig {
+    /// A location to a file representing the root CA certificate used by the server. If this is
+    /// set, core will attempt to use TLS when connecting to the server. If `client_tls_config` is
+    /// also set, then core will attempt to use mTLS.
+    pub server_root_ca_cert: PathBuf,
```

Comment:
We'll need to support reading certs from buffers instead of paths, we don't know how our users obtain their certificates (see example [here](https://temporalio.slack.com/archives/C01DKSMU94L/p1620409279021300?thread_ts=1620333054.020200&cid=C01DKSMU94L))

Reply:
Ah yeah, reading from a buffer makes perfect sense. What's the scenario where there's no root cert? A cert from a trusted issuer? Is that the case with our SaaS stuff?

Reply:
Yes, to answer my own question.